rust/processor/src/db/postgres/migrations/2025-01-08-171729_emojicoin_arena/up.sql

```diff
@@ -0,0 +1,256 @@
+    emojicoin_1_proceeds NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_quote NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_quote NUMERIC NOT NULL,
+
+    PRIMARY KEY (transaction_version, event_index)
+);
+
+CREATE TABLE arena_exit_events (
+    transaction_version BIGINT NOT NULL,
+    event_index BIGINT NOT NULL,
+    sender VARCHAR(66) NOT NULL,
+    entry_function VARCHAR(200),
+    transaction_timestamp TIMESTAMP NOT NULL,
+    inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+    "user" TEXT NOT NULL,
+    melee_id NUMERIC NOT NULL,
+    tap_out_fee NUMERIC NOT NULL,
+    emojicoin_0_proceeds NUMERIC NOT NULL,
+    emojicoin_1_proceeds NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_quote NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_quote NUMERIC NOT NULL,
+
+    PRIMARY KEY (transaction_version, event_index)
+);
+
+CREATE TABLE arena_swap_events (
+    transaction_version BIGINT NOT NULL,
+    event_index BIGINT NOT NULL,
+    sender VARCHAR(66) NOT NULL,
+    entry_function VARCHAR(200),
+    transaction_timestamp TIMESTAMP NOT NULL,
+    inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+    "user" TEXT NOT NULL,
+    melee_id NUMERIC NOT NULL,
+    quote_volume NUMERIC NOT NULL,
+    integrator_fee NUMERIC NOT NULL,
+    emojicoin_0_proceeds NUMERIC NOT NULL,
+    emojicoin_1_proceeds NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_0_exchange_rate_quote NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_base NUMERIC NOT NULL,
+    emojicoin_1_exchange_rate_quote NUMERIC NOT NULL,
+
+    PRIMARY KEY (transaction_version, event_index)
+);
+
+CREATE TABLE arena_vault_balance_update_events (
+    transaction_version BIGINT NOT NULL,
+    event_index BIGINT NOT NULL,
+    sender VARCHAR(66) NOT NULL,
+    entry_function VARCHAR(200),
+    transaction_timestamp TIMESTAMP NOT NULL,
+    inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+
+    new_balance NUMERIC NOT NULL,
+
+    PRIMARY KEY (transaction_version, event_index)
+);
+
+-- Derived data
+
+CREATE TABLE arena_positions (
+    "user" TEXT NOT NULL,
+    melee_id NUMERIC NOT NULL,
+    open BOOL NOT NULL,
+    emojicoin_0_balance NUMERIC NOT NULL,
+    emojicoin_1_balance NUMERIC NOT NULL,
+    profits NUMERIC NOT NULL,
+    losses NUMERIC NOT NULL,
+
+    PRIMARY KEY ("user", melee_id)
+);
+
+CREATE TABLE arena_leaderboard_history (
+    "user" TEXT NOT NULL,
+    melee_id NUMERIC NOT NULL,
+    profits NUMERIC NOT NULL,
+    losses NUMERIC NOT NULL,
+
+    PRIMARY KEY ("user", melee_id)
+);
+
+-- Functions
+
+CREATE OR REPLACE FUNCTION arena_leaderboard() RETURNS TABLE(
+    "user" TEXT,
+    open BOOL,
+    emojicoin_0_balance NUMERIC,
+    emojicoin_1_balance NUMERIC,
+    profits NUMERIC,
+    losses NUMERIC,
+    pnl NUMERIC
```

Comment:
Are we able to sort on both total pnl as a percent and as a net value?

For example the user who has 2000% gains but only put in 100 to start might win one leaderboard

Whereas the user who doubled 2000000 APT might have higher net PnL